src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyout.cs

```diff
@@ -80,31 +82,314 @@ public IList<MenuFlyoutItemBase> Items
 			set => SetValue(MenuFlyoutPresenterStyleProperty, value);
 		}
 
-		public static global::Windows.UI.Xaml.DependencyProperty MenuFlyoutPresenterStyleProperty { get; } =
-		Windows.UI.Xaml.DependencyProperty.Register(
+		public static DependencyProperty MenuFlyoutPresenterStyleProperty { get; } =
+		DependencyProperty.Register(
 			"MenuFlyoutPresenterStyle",
-			typeof(global::Windows.UI.Xaml.Style),
-			typeof(global::Windows.UI.Xaml.Controls.MenuFlyout),
-			new FrameworkPropertyMetadata(null));
+			typeof(Style),
+			typeof(MenuFlyout),
+			new PropertyMetadata(null, (s, e) => (s as MenuFlyout).OnPropertyChanged2(s, e)));
+
+		public void ShowAt(UIElement targetElement, Point point)
+		{
+			ShowAtCore((FrameworkElement)targetElement);
+		}
 
-		public void ShowAt(global::Windows.UI.Xaml.UIElement targetElement, global::Windows.Foundation.Point point)
+		private void OnPropertyChanged2(DependencyObject s, DependencyPropertyChangedEventArgs e)
 		{
-			base.ShowAt((FrameworkElement)targetElement);
+			if (e.Property == MenuFlyoutPresenterStyleProperty &&
+				GetPresenter() != null)
+			{
+				SetPresenterStyle(GetPresenter(), e.NewValue as Style);
+			}
 		}
 
 		protected override Control CreatePresenter()
 		{
-			_presenter = new MenuFlyoutPresenter();
+			MenuFlyoutPresenter presenter = new MenuFlyoutPresenter();
+			Style style;
+
+			presenter.SetParentMenuFlyout(this);
+
+			style = MenuFlyoutPresenterStyle;
+			SetPresenterStyle(presenter, style);
+
+			return presenter;
+		}
+
+		private protected override void ShowAtCore(FrameworkElement pPlacementTarget)
+		{
+			base.ShowAtCore(pPlacementTarget);
+
+			if (m_openWindowed)
+			{
+				// UNO TODO
+				// SetIsWindowedPopup();
+			}
+			else
+			{
+				m_openWindowed = true;
+			}
+
+			var placementTarget = pPlacementTarget;
+			CacheInputDeviceTypeUsedToOpen(placementTarget);
+		}
+
+		// Raise Opening event.
+		private protected override void OnOpening()
+		{
+			// Update the TemplateSettings as it is about to open.
+			Control presenter = GetPresenter();
+			MenuFlyoutPresenter menuFlyoutPresenter = presenter as MenuFlyoutPresenter;
+
+			if (menuFlyoutPresenter != null)
+			{
+				IMenu parentMenu = (this as IMenu).ParentMenu as IMenu;
+
+				(menuFlyoutPresenter as IMenuPresenter).OwningMenu = parentMenu != null ? parentMenu : this;
+				menuFlyoutPresenter.UpdateTemplateSettings();
+			}
+
+			// Reset the presenter's ItemsSource.  Since Items is not an IObservableVector, we don't
+			// automatically respond to changes within the vector.  Clearing the property when the presenter
+			// unloads and resetting it before we reopen ensures any changes to Items are reflected
+			// when the MenuFlyoutPresenter shows.  It also allows sharing of MenuFlyouts; since MenuFlyoutItemBases
+			// are UIElements they must be unparented when leaving the tree before they can be inserted elsewhere.
+			menuFlyoutPresenter.ItemsSource = m_tpItems;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['menuFlyoutPresenter' is null on at least one execution path.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)